File: Example/ObjC/ExampleViewController.m
//
//  Created by Stefan Nebel on 28.06.20.
//  Copyright © 2020 Best-iOS-Apps. All rights reserved.
//

#import "ExampleViewController.h"

@interface ExampleViewController ()
@property ExampleViewModel *viewModel;
@end

@implementation ExampleViewController

- (nonnull instancetype)initWithModel:(ExampleViewModel *)viewModel {
  self = [super init];
  if (self) {
    self.viewModel = viewModel;
  }
  return self;
}

- (void)viewDidLoad {
  [super viewDidLoad];

  for (NSString *content in _viewModel.deviceInfos) {
    NSLog(@"%@", content);
  }
}


@end

File: Example/ExampleViewController.swift
//
//  Created by Stefan Nebel on 14.06.20.
//  Copyright © 2020 Best-iOS-Apps. All rights reserved.
//

import UIKit
import BIASystemKit

class ExampleViewController: UIViewController {

  let viewModel: ViewModel

  init(viewModel: ViewModel) {
    self.viewModel = viewModel
    super.init(nibName: nil, bundle: nil)
  }

  required init?(coder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }

  override func viewDidLoad() {
    super.viewDidLoad()

    print(UIDevice.current.formattedName)
  }
}
